Jenkinsfile

```
@@ -78,13 +79,25 @@ pipeline {
                                 -e POSTGRES_DB=oqt \
                                 -e POSTGRES_USER=oqt \
                                 -e POSTGRES_PASSWORD=oqt""") { c ->
-              WORKERS.inside("--network ${n} -e POSTGRES_HOST=${POSTGRES_HOST} -e POSTGRES_PORT=${POSTGRES_PORT}") {
+              WORKERS_CI.inside("--network ${n} -e POSTGRES_HOST=${POSTGRES_HOST} -e POSTGRES_PORT=${POSTGRES_PORT}") {
                 // wait for database to be ready
                 timeout(time: 30, unit: 'SECONDS') {
                   sh 'while ! pg_isready --host ${POSTGRES_HOST} --port ${POSTGRES_PORT}; do sleep 5; done'
                 }
                 // run pytest
-                sh 'cd ${WORK_DIR} && ${POETRY_RUN} pytest tests'
+                sh 'cd ${WORK_DIR} && ${POETRY_RUN} pytest --cov=ohsome_quality_analyst --cov-report=xml tests'
+                // replace absolute dir in the coverage file with actually used dir for sonar-scanner
+                sh "sed -i \"s#${WORK_DIR}#${WORKSPACE}/${MODULE_DIR}#g\" ${WORK_DIR}/coverage.xml"
+                // run static analysis with sonar-scanner
+                def scannerHome = tool 'SonarScanner 4';
+                withSonarQubeEnv('sonarcloud GIScience/ohsome') {
+                  sh "${scannerHome}/bin/sonar-scanner " +
+                    "-Dsonar.branch.name=${env.BRANCH_NAME} " +
+                    "-Dsonar.python.coverage.reportPaths=${WORK_DIR}/coverage.xml"
+                }
+                sh 'cd ${WORK_DIR} && ${POETRY_RUN} black --check --diff --no-color .'
```

maybe add an inline comment before this line so it's clear that this doesn't belong to `// run static analysis with sonar-scanner` anymore.

```suggestion
                // run other static code analysis
                sh 'cd ${WORK_DIR} && ${POETRY_RUN} black --check --diff --no-color .'
```